contracts/src/systems/order/take_fungible_order.cairo

```diff
@@ -232,361 +233,301 @@ mod TakeFungibleOrder {
+        set!(world, (Position { x: 900_000, y: 100_000, entity_id: taker_caravan_id.into()}));
+        set!(world, (Capacity { weight_gram: 10_000, entity_id: taker_caravan_id.into()}));
+        set!(world, (Movable { sec_per_km: 10, blocked: false, entity_id: taker_caravan_id.into()}));
+        // attach caravan to the taker order
+        let taker_caravan_key_arr = array![taker_order_id.into(), taker_id.into()];
+        let taker_caravan_key = poseidon_hash_span(taker_caravan_key_arr.span());
+        set!(world, (Caravan { caravan_id: taker_caravan_id.into(), entity_id: taker_caravan_key }));
+
+
+
+        // taker takes trade
+        starknet::testing::set_contract_address(
+            contract_address_const::<'taker'>()
+        );
+        let mut calldata = array![];
+        Serde::serialize(@taker_id, ref calldata);
+        Serde::serialize(@trade_id, ref calldata);
+        world.execute('TakeFungibleOrder', calldata);
+
+
+
+        // taker wood balance should be 500
+        let taker_wood_resource = get!(world, (taker_id, ResourceTypes::WOOD), Resource);
+        assert(taker_wood_resource.balance == 500, 'resource balance should be 500'); // 600 - 100
+
+        // taker stone balance should be 400
+        let taker_stone_resource = get!(world, (taker_id, ResourceTypes::STONE), Resource);
+        assert(taker_stone_resource.balance == 400, 'resource balance should be 400'); // 600 - 200
+
+
+        // status should be accepted
+        let status = get!(world, trade_id, Status);
+        assert(status.value == 1, 'status should be accepted');
+
+        
+        // verify arrival time and position of maker order
+        let maker_order_arrival_time = get!(world, maker_order_id, ArrivalTime);
+        let maker_order_position = get!(world, maker_order_id, Position);
+        assert(maker_order_arrival_time.arrives_at == 220, 'arrival time should be 220');
+        assert(maker_order_position.x == 60, 'position x should be 60');
+        assert(maker_order_position.y == 70, 'position y should be 70');
+        // verify arrival time of maker caravan
+        let maker_caravan_arrival_time = get!(world, maker_caravan_id, ArrivalTime);
+        assert(maker_caravan_arrival_time.arrives_at == (220 * 2), 'arrival time should be 440');
+
+ 
+
+        // verify arrival time and position of taker order
+        let taker_order_arrival_time = get!(world, taker_order_id, ArrivalTime);
+        let taker_order_position = get!(world, taker_order_id, Position);
+        assert(taker_order_arrival_time.arrives_at == 900, 'arrival time should be 900');
+        assert(taker_order_position.x == 45, 'position x should be 45');
+        assert(taker_order_position.y == 50, 'position y should be 50');
+        // verify arrival time of taker caravan
+        let taker_caravan_arrival_time = get!(world, taker_caravan_id, ArrivalTime);
+        assert(taker_caravan_arrival_time.arrives_at == (900 * 2), 'arrival time should be 1800');
```

Comment:
if both realms have a caravan with the same speed, the taker and maker caravans should arrive at the same time

Reply:
Fixed